### ISSUE TITLE
D400 should accept question mark as a sentence ender

### ISSUE THREAD

**Issue by warsaw** [astral-sh/ruff#8020]
`D400` complains when the first line of a docstring ends in a `?`.   However, it seems natural when documenting a predicate function that the first line of the docstring poses a question.  `D400` should accept `?` as a sentence ender in those cases.

```python
# example.py
def is_a_thing(thing: int) -> bool:
    """Is this a thing?"""
    return bool(thing)
```

```toml
# ruff.toml
select = ['D400']
```

```shell
$ ruff check example.py 
example.py:2:5: D400 First line should end with a period
Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).
```

**Comment by charliermarsh** [comment]
Following pydocstyle, I think there's a different rule for this (`D415`). Could you disable `D400` and enable `D415` instead?

**Comment by warsaw** [comment]
Oh interesting.  I tried to boil it down to an easy reproducer, but actually, in my production code (which I can't share), I have `select = ['D']` rather than specifically `D400`.  Maybe there's a weird interaction going on here?

Yup, if I change the `ruff.toml` to just:

```toml
# ruff.toml
select = ['D']
```

I get

```
warning: `one-blank-line-before-class` (D203) and `no-blank-line-before-class` (D211) are incompatible. Ignoring `one-blank-line-before-class`.
warning: `multi-line-summary-first-line` (D212) and `multi-line-summary-second-line` (D213) are incompatible. Ignoring `multi-line-summary-second-line`.
foo.py:1:1: D100 Missing docstring in public module
foo.py:2:5: D400 [*] First line should end with a period
Found 2 errors.
[*] 1 potentially fixable with the --fix option.
```

**Comment by charliermarsh** [comment]
Do you have a convention set? Like:

```toml
[tool.ruff.pydocstyle]
convention = "google"
```

That effectively acts as a filter on the select.

**Comment by warsaw** [comment]
Yes, I'm using `convention = 'pep257'`.

**Comment by charliermarsh** [comment]
I think the solution here is to fix #2606, then suggest using `D415` in lieu of `D400`. Right now, it's impossible to enable `D415` alongside that convention.

**Comment by charliermarsh** [comment]
Okay, in the next release, you should be able to add:

```toml
select = ["D", "D415"]
ignore = ["D400"]

[pydocstyle]
convention = "pep257"
```

To enable the PEP 257 convention, but swapping out D400 for D415.
